From 101934f29e6b76931b1499adc19ae7f7a976789d Mon Sep 17 00:00:00 2001
From: Lukas Slebodnik <lslebodn@redhat.com>
Date: Tue, 14 Aug 2018 08:20:57 +0000
Subject: [PATCH 1/4] BUILD: Fix issue with installation of libsss_secrets
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

libsss_secret.so is linked with libsss_util.so therefore it shoudl be
added into pkglib_LTLIBRARIES after libsss_util.so.
Otherwise there can failure in linking phase.

libtool: warning: relinking 'libsss_secrets.la'
libtool: install: (cd /home/build/sssd/ci-build-debug/intg/bld; /bin/sh
  "/home/build/sssd/ci-build-debug/intg/bld/libtool"  --tag CC
  --mode=relink gcc -Wall -Wshadow -Wstrict-prototypes -Wpointer-arith
  -Wcast-qual -Wcast-align -Wwrite-strings -Wundef
  -Werror-implicit-function-declaration -Winit-self -Wmissing-include-dirs
  -fno-strict-aliasing -std=gnu99 -O2 -g -g3 -O2 -Werror
  -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE
  -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE
  -avoid-version -o libsss_secrets.la -rpath
  /tmp/sssd-intg.l7nl5pgb/lib/sssd
  src/util/secrets/libsss_secrets_la-secrets.lo
  src/util/secrets/libsss_secrets_la-config.lo -ltalloc -lldb
  libsss_crypt.la libsss_debug.la libsss_util.la )
libtool: relink: gcc -shared  -fPIC -DPIC
  src/util/secrets/.libs/libsss_secrets_la-secrets.o
  src/util/secrets/.libs/libsss_secrets_la-config.o   -Wl,-rpath
  -Wl,/tmp/sssd-intg.l7nl5pgb/lib/sssd -ltalloc -lldb
  -L/tmp/sssd-intg.l7nl5pgb/lib/sssd -lsss_crypt -lsss_debug -lsss_util
  -O2 -g -g3 -O2   -Wl,-soname -Wl,libsss_secrets.so -o
  .libs/libsss_secrets.so
/usr/bin/ld: cannot find -lsss_util
collect2: error: ld returned 1 exit status
libtool:   error: error: relink 'libsss_secrets.la' with the above
command before installing it

Merges: https://pagure.io/SSSD/sssd/pull-request/3808

Reviewed-by: Pavel Březina <pbrezina@redhat.com>
---
 Makefile.am | 43 +++++++++++++++++++++----------------------
 1 file changed, 21 insertions(+), 22 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index d313957..a2d8ea4 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1209,28 +1209,6 @@ libsss_iface_sync_la_LDFLAGS = \
     -avoid-version \
     $(NULL)
 
-if BUILD_WITH_LIBSECRET
-pkglib_LTLIBRARIES += libsss_secrets.la
-
-libsss_secrets_la_SOURCES = \
-    src/util/secrets/secrets.c \
-    src/util/secrets/config.c \
-    $(NULL)
-libsss_secrets_la_CFLAGS = \
-    $(AM_CFLAGS) \
-    $(NULL)
-libsss_secrets_la_LIBADD = \
-    $(TALLOC_LIBS) \
-    $(LDB_LIBS) \
-    libsss_crypt.la \
-    libsss_debug.la \
-    libsss_util.la \
-    $(NULL)
-libsss_secrets_la_LDFLAGS = \
-    -avoid-version \
-    $(NULL)
-endif
-
 pkglib_LTLIBRARIES += libsss_util.la
 libsss_util_la_SOURCES = \
     src/confdb/confdb.c \
@@ -1314,6 +1292,27 @@ libsss_util_la_LIBADD += stap_generated_probes.lo
 endif
 libsss_util_la_LDFLAGS = -avoid-version
 
+if BUILD_WITH_LIBSECRET
+pkglib_LTLIBRARIES += libsss_secrets.la
+libsss_secrets_la_SOURCES = \
+    src/util/secrets/secrets.c \
+    src/util/secrets/config.c \
+    $(NULL)
+libsss_secrets_la_CFLAGS = \
+    $(AM_CFLAGS) \
+    $(NULL)
+libsss_secrets_la_LIBADD = \
+    $(TALLOC_LIBS) \
+    $(LDB_LIBS) \
+    libsss_crypt.la \
+    libsss_debug.la \
+    libsss_util.la \
+    $(NULL)
+libsss_secrets_la_LDFLAGS = \
+    -avoid-version \
+    $(NULL)
+endif
+
 pkglib_LTLIBRARIES += libsss_semanage.la
 libsss_semanage_la_CFLAGS = \
     $(AM_CFLAGS) \
-- 
2.9.5

